hedera-node/src/main/java/com/hedera/services/txns/crypto/validators/AllowanceChecks.java

```diff
@@ -83,11 +82,8 @@ ResponseCodeEnum validateSerialNums(
 			final List<Long> serialNums,
 			final Token token,
 			final ReadOnlyTokenStore tokenStore) {
-		if (hasRepeatedSerials(serialNums)) {
-			return REPEATED_SERIAL_NUMS_IN_NFT_ALLOWANCES;
-		}
-
-		for (var serial : serialNums) {
+		final var serialsSet = new HashSet<>(serialNums);
```

Comment:
Is this to save us from checking repeated serials ??
If the limit is just 20, I think we can skip creating a new Hashset and just iterate over the list

Reply:
I thought , we will avoid many unnecessary iterations, if the limit needs to be increased in the future.